psycopg/psycopg/pq/_pq_ctypes.py

```diff
@@ -593,6 +598,135 @@ def PQsendClosePortal(pgconn: PGconn_struct, name: str) -> int:
 
 # 33.6. Canceling Queries in Progress
 
+_PQcancelCreate = None
+_PQcancelStart = None
+_PQcancelBlocking = None
+_PQcancelPoll = None
+_PQcancelStatus = None
+_PQcancelSocket = None
+_PQcancelErrorMessage = None
+_PQcancelReset = None
+_PQcancelFinish = None
+
+if libpq_version >= 170000:
+    _PQcancelCreate = pq.PQcancelCreate
+    _PQcancelCreate.argtypes = [PGconn_ptr]
+    _PQcancelCreate.restype = PGcancelConn_ptr
+
+    _PQcancelStart = pq.PQcancelStart
+    _PQcancelStart.argtypes = [PGcancelConn_ptr]
+    _PQcancelStart.restype = c_int
+
+    _PQcancelBlocking = pq.PQcancelBlocking
+    _PQcancelBlocking.argtypes = [PGcancelConn_ptr]
+    _PQcancelBlocking.restype = c_int
+
+    _PQcancelPoll = pq.PQcancelPoll
+    _PQcancelPoll.argtypes = [PGcancelConn_ptr]
+    _PQcancelPoll.restype = c_int
+
+    _PQcancelStatus = pq.PQcancelStatus
+    _PQcancelStatus.argtypes = [PGcancelConn_ptr]
+    _PQcancelStatus.restype = c_int
+
+    _PQcancelSocket = pq.PQcancelSocket
+    _PQcancelSocket.argtypes = [PGcancelConn_ptr]
+    _PQcancelSocket.restype = c_int
+
+    _PQcancelErrorMessage = pq.PQcancelErrorMessage
+    _PQcancelErrorMessage.argtypes = [PGcancelConn_ptr]
+    _PQcancelErrorMessage.restype = c_char_p
+
+    _PQcancelReset = pq.PQcancelReset
+    _PQcancelReset.argtypes = [PGcancelConn_ptr]
+    _PQcancelReset.restype = None
+
+    _PQcancelFinish = pq.PQcancelFinish
+    _PQcancelFinish.argtypes = [PGcancelConn_ptr]
+    _PQcancelFinish.restype = None
+
+
+def PQcancelCreate(pgconn: PGconn_struct) -> PGcancelConn_struct:
+    if not _PQcancelCreate:
+        raise NotSupportedError(
+            "PQcancelCreate requires libpq from PostgreSQL 17,"
+            f" {libpq_version} available instead"
+        )
+    return _PQcancelCreate(pgconn)  # type: ignore[no-any-return]
```

Comment:
Actually, the "raise unsupported" was becoming a nuisance, so in the same MR I've added further refactoring to remove that pattern. WDYT?

Reply:
> How about specifying the return type as in https://github.com/psycopg/psycopg/pull/768?

I have mixed feelings about that; I think the explicit `# type: ignore[no-any-return]` indicate more clearly that there is a limitation somewhere (on our side or in mypy) and that it might go away someday. However I can live with the approach you suggest, especially if it's more consistent with the rest of the code base (which seems true, as evidenced by `git grep no-any-return` vs `git grep "rv: "`).

Reply:
Hm, reading the next commits in #768 makes my point vanish; so no objection at all :)

Reply:
I rebased on now-master so the code now looks like #768.